portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -60,12 +60,22 @@ private int getOffset(HttpServletRequest request) {
     return getNonNegativeIntParameter(request, "offset", DEFAULT_COMMENT_OFFSET);
   }
 
+  /**
+   * Returns the list of comments translated.
+   */
+  private List<Comment> loadAndTranslateComments(HttpServletRequest request) {
+    List<Comment> comments = CommentsStore.load(getLimit(request), getOffset(request));
+    String languageCode = getParameter(request, "language", "en");
```

Comment:
Try extracting all constants into actual class constants. E.g. `private static final PARAM_LANGUAGE = "langauge";`
This also helps with memory usage, even though marginally, and makes it clear to a reader that those are really constant. Additionally they can be reused easier where needed.

Reply:
Have you considered what happens if a language is not supported?

Reply:
I will add the handling of this case.